src/AddIns/Uno.UI.Lottie/ts/Uno.UI.Lottie.ts

```diff
@@ -17,100 +17,114 @@ namespace Uno.UI {
 	export class Lottie {
 		private static _player: LottiePlayer;
 		private static _runningAnimations: { [id: number]: RunningLottieAnimation } = {};
+		private static _numberOfFrames: number;
 
 		public static setAnimationProperties(newProperties: LottieAnimationProperties): string {
 			const elementId = newProperties.elementId;
 
-			this.withPlayer(p => {
-				let currentAnimation = this._runningAnimations[elementId];
+			Lottie.withPlayer(p => {
+				let currentAnimation = Lottie._runningAnimations[elementId];
 
-				if (!currentAnimation || this.needNewPlayerAnimation(currentAnimation.properties, newProperties)) {
+				if (!currentAnimation || Lottie.needNewPlayerAnimation(currentAnimation.properties, newProperties)) {
 					// Here we need a new player animation
 					// (some property changes required a new animation)
-					currentAnimation = this.createAnimation(newProperties);
+					currentAnimation = Lottie.createAnimation(newProperties);
 				}
 
-				this.updateProperties(currentAnimation, newProperties);
+				Lottie.updateProperties(currentAnimation, newProperties);
 			});
 
 			return "ok";
 		}
 
 		public static stop(elementId: number): string {
-			this.withPlayer(p => {
-				this._runningAnimations[elementId].animation.stop();
+			Lottie.withPlayer(p => {
+				const a = Lottie._runningAnimations[elementId].animation;
+				a.stop();
+				Lottie.raiseState(a);
 			});
 
 			return "ok";
 		}
 
-		public static play(elementId: number, looped: boolean): string {
-			this.withPlayer(p => {
-				const a = this._runningAnimations[elementId].animation;
+		public static play(elementId: number, fromProgress: number, toProgress: number, looped: boolean): string {
+			Lottie.withPlayer(p => {
+				const a = Lottie._runningAnimations[elementId].animation;
 				a.loop = looped;
-				a.play();
+
+				var fromFrame = fromProgress * Lottie._numberOfFrames;
+				var toFrame = toProgress * Lottie._numberOfFrames;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Identifier 'toFrame' is never reassigned; use 'const' instead of 'var'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5436651)